src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx

```diff
@@ -0,0 +1,307 @@
+  hasMore,
+  isLoadingMore,
+  currentScrollRef,
+  alternateOldestViewEnabled,
+  commentsOrderBy,
+  comments,
+  newCommentsLength,
+}) => {
+  const [local, setLocal] = useLocal<
+    AllCommentsTabVirtualizedCommentsLocal
+  >(graphql`
+    fragment AllCommentsTabVirtualizedCommentsLocal on Local {
+      showLoadAllCommentsButton
+      oldestFirstNewCommentsToShow
+      totalCommentsLength
+      zKeyClickedLoadAll
+      addACommentButtonClicked
+      keyboardShortcutsConfig {
+        key
+        source
+        reverse
+      }
+    }
+  `);
+
+  // We need to know if the Load all button has been clicked to help determine whether
+  // to display the Load all button or not.
+  const [
+    loadAllButtonHasBeenClicked,
+    setLoadAllButtonHasBeenClicked,
+  ] = useState(false);
+
+  // Initial comments are used to determine whether the Load all comments button should be
+  // displayed or not.
+  const [initialComments, setInitialComments] = useState<null | {
+    length: number;
+    hasMore: boolean;
+  }>(null);
+
+  useEffect(() => {
+    setLocal({ totalCommentsLength: comments.length });
+  }, [comments.length, setLocal]);
+
+  // This determines if there are more comments to display than the initial 20.
+  // It also takes into account the initial comments loaded since if we start
+  // with fewer than 20, we will never want to display load all as new comments
+  // come in.
+  const moreCommentsForLoadAll = useMemo(() => {
+    return (
+      comments.length > NUM_INITIAL_COMMENTS &&
+      ((initialComments && initialComments.length > NUM_INITIAL_COMMENTS) ||
+        (initialComments &&
+          initialComments.length === NUM_INITIAL_COMMENTS &&
+          initialComments.hasMore))
+    );
+  }, [comments.length, initialComments]);
+
+  // We determine whether to display the Load all comments button based on whether:
+  // 1. If there are more comments to display than 20 AND fewer than 20 weren't initially loaded.
+  // 2. Don't display if Z key has clicked the Load all button to open it and go to next unseen.
+  // 3. Display if the Add a comment button has been clicked in alternate oldest view.
+  // 4. If alternate oldest view enabled, don't display if Load all button has been clicked;
+  // otherwise, default to the admin configuration.
+  // 5. Last, we check the admin configuration and displayed based on that and whether or not
+  // the Load all button has already been clicked.
+  const displayLoadAllButton = useMemo(() => {
+    if (moreCommentsForLoadAll) {
+      if (local.zKeyClickedLoadAll) {
+        return false;
+      }
+      if (local.addACommentButtonClicked) {
+        return true;
+      }
+      if (alternateOldestViewEnabled) {
+        return loadAllButtonHasBeenClicked ? false : !settings.loadAllComments;
+      }
+      if (!settings.loadAllComments) {
+        return !loadAllButtonHasBeenClicked;
+      } else {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  }, [
```

Comment:
This is perfect for handling this complexity. Very easy to jump in with console logging or break points. Thank you!

Reply:
Great! I was also able to simplify this with the updates to not have a configuration setting for `loadAllComments` anymore and with alternate oldest view just scrolling down to the last comment when clicking `Add a comment` now.